extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -618,30 +692,61 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
      * @param value the geometry
      */
     fun geometry(value: Geometry) = apply {
-      rawGeoJson = value
-      val propertyValue = PropertyValue("data", "null")
-      properties[propertyValue.propertyName] = propertyValue
+      setGeoJson(value)
+    }
+
+    private fun setGeoJson(geoJson: GeoJson) {
+      this.geoJson = geoJson
+      if (directSetterEnabled) {
+        data = null
+      } else {
+        val propertyValue = PropertyValue("data", "null")
+        properties[propertyValue.propertyName] = propertyValue
+      }
     }
 
     /**
      * Initialize GeoJsonSource builder with default data to allow empty data source.
      */
     init {
-      this.data("")
+      if (!directSetterEnabled) {
+        this.data("")
+      }
     }
 
     /**
      * Build the GeoJsonSource.
      *
      * @return the GeoJsonSource
      */
-    fun build() = GeoJsonSource(this, rawGeoJson)
+    fun build(): GeoJsonSource {
+      if (directSetterEnabled) {
+        // we need empty initial data when creating the geojson (as we don't set it in constructor) -
```

Comment:
```suggestion
        // set default data to allow empty data source.
```
the same comment is added to the builder